tavern_server/src/db_tables.sql

```diff
@@ -0,0 +1,861 @@
+	req_attr		Attribute
+		NOT NULL,
+	score			INT
+		NOT NULL
+);
+
+--this table creates a many to many realationship internally between feats.
+--any given feat can require a number of other feats to be taken,
+--and a given feat can be required by any number of other feats.
+--this table accomplishes that.
+CREATE TABLE IF NOT EXISTS FeatRequirements (
+	feat_id			UUID REFERENCES Feats
+		NOT NULL,
+	required_feat		UUID REFERENCES Feats
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS CharacterFeats (
+	char_id			UUID REFERENCES Characters
+		NOT NULL,
+	feat_id			UUID REFERENCES Feats
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS RacialFeats (
+	race_id			UUID REFERENCES Races
+		NOT NULL,
+	feat_id			UUID REFERENCES Feats
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS ClassFeats (
+	class_id		UUID REFERENCES Classes,
+		NOT NULL
+	feat_id			UUID REFERENCES Feats
+		NOT NULL
+);
+
+--Features tables
+CREATE TABLE IF NOT EXISTS Features (
+	feature_id		UUID PRIMARY KEY,
+	description		TEXT
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS ClassFeatures (
+	class_id		UUID REFERENCES Feats
+		NOT NULL,
+	feature_id		UUID REFERENCES Features
+		NOT NULL,
+	is_default		BOOLEAN
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS CharacterFeatures (
+	char_id			UUID REFERENCES Characters
+		NOT NULL,
+	feature_id		UUID REFERENCES Features
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS SubclassFeatures (
+	subclass_id		UUID REFERENCES Subclasses
+		NOT NULL,
+	feature_id		UUID REFERENCES Features
+		NOT NULL
+);
+
+--class proficiencies tables
+CREATE TABLE IF NOT EXISTS ClassWeaponsNotProficient (
+	class_id		UUID REFERENCES Classes
+		NOT NULL,
+	weapon_classes		WeaponClass[]
+);
+
+CREATE TABLE IF NOT EXISTS ClassArmorsNotProficient (
+	class_id		UUID REFERENCES Classes
+		NOT NULL,
+	armor_classes		ArmorClass[]
+);
+
+--Spells tables
+CREATE TABLE IF NOT EXISTS Spells (
+	spell_id		UUID PRIMARY KEY,
+
+	name			TEXT
+		NOT NULL,
+	level			INT
+		NOT NULL
+		CHECK (level >= 0),
+	school			MagicSchool
+		NOT NULL,
+
+	casting_time		INT
+		NOT NULL
+		CHECK (casting_time >= 0),
+	range			SpellRange
+		NOT NULL,
+	area			TEXT
+		NOT NULL,
+	duration_per_level	INT
+		NOT NULL
+		CHECK (duration_per_level > 0),
+	saving_throw		SaveThrow,
+	spell_resistance	BOOLEAN
+		NOT NULL,
+	description		TEXT
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS CharacterSpells (
+	char_id			UUID REFERENCES Characters
+		NOT NULL,
+	spell_id		UUID REFERENCES Spells
+		NOT NULL,
+	casts			INT
+		NOT NULL
+		CHECK (casts >= 0)
+);
+
+CREATE TABLE IF NOT EXISTS SubclassSpells (
+	subclass_id		UUID REFERENCES Subclasses,
+	spell_id		UUID REFERENCES Spells,
+	casts			INT,
+	req_level		INT
+);
+
+CREATE TABLE IF NOT EXISTS DomainSpells (
+	domain_id		UUID REFERENCES Domains,
+	spell_id		UUID REFERENCES Spells,
+	casts			INT
+);
+
+CREATE TABLE IF NOT EXISTS SpellComponents (
+	spell_id		UUID REFERENCES Spells
+		NOT NULL,
+	item_id			UUID REFERENCES Items,
+
+	amount			INT,
```

Comment:
What does `amount` mean in this context? Is it not required?

Reply:
renamed to item_amount - it is not required if the component is Verbal or Somatic.